modules/cfe_testcase/src/resource_id_misc_test.c

```diff
@@ -0,0 +1,161 @@
+    UtAssert_UINT32_EQ(CFE_ResourceId_ToInteger(ResourceId2), Id2);
+    UtAssert_UINT32_EQ(ResourceId1, ResourceId1Exp);
+    UtAssert_UINT32_EQ(ResourceId2, ResourceId2Exp);
+}
+
+void TestIsDefined(void)
+{
+    UtPrintf("Testing: CFE_ResourceId_IsDefined");
+    CFE_ResourceId_t ResourceId1 = 1;
+    CFE_ResourceId_t ResourceId2 = 999;
+    UtAssert_BOOL_TRUE(CFE_ResourceId_IsDefined(ResourceId1));
+    UtAssert_BOOL_TRUE(CFE_ResourceId_IsDefined(ResourceId2));
+    UtAssert_BOOL_FALSE(CFE_ResourceId_IsDefined(CFE_RESOURCEID_UNDEFINED));
+}
+
+void TestGetBaseSerial(void)
+{
+    UtPrintf("Testing: CFE_ResourceId_GetBase, CFE_ResourceId_GetSerial");
+    CFE_ES_AppId_t     AppId;
+    CFE_ES_LibId_t     LibId;
+    CFE_ES_TaskId_t    TaskId;
+    CFE_ES_CounterId_t CounterId;
+    CFE_ES_MemHandle_t PoolId;
+    CFE_ES_CDSHandle_t CDSHandleId;
+    const char *       LibName     = "ASSERT_LIB";
+    const char *       CounterName = "TEST_COUNTER";
+    int8               Pool[1024];
+    size_t             CDSBlockSize = 10;
+    const char *       CDSName      = "TEST_CDS";
+
+    /* Referenced from cfe_core_resourceid_basevalues.h */
+    int TASKID_BASE     = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_OS_TASK);
+    int APPID_BASE      = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 1);
+    int LIBID_BASE      = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 2);
+    int COUNTID_BASE    = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 3);
+    int POOLID_BASE     = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 4);
+    int CDSBLOCKID_BASE = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 5);
+
+    /* App ID */
+    UtAssert_INT32_EQ(CFE_ES_GetAppID(&AppId), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(AppId), APPID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(AppId), AppId - APPID_BASE);
+    /* Lib ID */
+    UtAssert_INT32_EQ(CFE_ES_GetLibIDByName(&LibId, LibName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(LibId), LIBID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(LibId), LibId - LIBID_BASE);
+    /* Task ID */
+    UtAssert_INT32_EQ(CFE_ES_GetTaskID(&TaskId), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(TaskId), TASKID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(TaskId), TaskId - TASKID_BASE);
+    /* Counter ID */
+    UtAssert_UINT32_EQ(CFE_ES_RegisterGenCounter(&CounterId, CounterName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(CounterId), COUNTID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(CounterId), CounterId - COUNTID_BASE);
+    /* Pool ID */
+    UtAssert_INT32_EQ(CFE_ES_PoolCreate(&PoolId, Pool, sizeof(Pool)), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(PoolId), POOLID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(PoolId), PoolId - POOLID_BASE);
+    /* CDS Block Id */
+    CFE_ES_RegisterCDS(&CDSHandleId, CDSBlockSize, CDSName);
```

Comment:
Need to have this check for CFE_SUCCESS or CFE_ES_CDS_ALREADY_EXISTS. You could also check that it's status is >=CFE_SUCCESS

Reply:
@jphickey That macro appears to not be available to the functional tests. Is there a way to include it (I tried to include "ut_support.h") or is there a similar macro that is available that I can use?

Reply:
That macro is part of https://github.com/nasa/cFE/pull/1790 it's not in yet

Reply:
CFE_UtAssert_STATUS_OK is what you want to use and is in master now.